datalad/support/gitrepo.py

```diff
@@ -3546,8 +3546,8 @@ def _get_cache_key(label, paths, ref, untracked=None):
                 if to_state_r['type'] != 'dataset':
                     # no change in git record, and no change on disk
                     props = dict(
-                        state='clean' if f.exists() or \
-                              f.is_symlink() else 'deleted',
+                        state='clean' if to is not None or (f.exists() or \
+                              f.is_symlink()) else 'deleted',
```

Comment:
Okay, so at this point we know that the reported object ids for this file are identical in the `to` and `from` records.  If `to` is None, we're comparing to the working tree and a deleted file will still have an identical id, so we need to check whether the file is gone before declaring it clean.  This working tree check is irrelevant and wrong if `to` is a ref.

Makes sense.

Reply:
Thx, that is indeed the reasoning. I will take the liberty and place it into a comment ;-)

Reply:
Ah, no need!   I was just working through it.

Reply:
But I see you already did it, which is also ok :)

Reply:
Done nevertheless!